Core/PixelFiring.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  PixelFiring.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Networking
+
+public protocol PixelFiring {
+
+    static func fire(_ pixel: Pixel.Event,
```

Comment:
Had to make first param unnamed to avoid name ambiguity, but I think API still makes sense, only downside is that when using `Pixel` directly there are both variants available in autocomplete. In the end I believe we should be using protocol exclusively.